numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -394,6 +399,70 @@ def test_round_f8(self):
             compiled[1, 1](ary, i)
             self.assertEquals(ary[0], round(i))
 
+    def test_round_to_f4(self):
```

Comment:
I'm not sure the unit tests trigger this branch https://github.com/numba/numba/pull/5741/files#diff-a3ad32c82e29f57220d54a522266e48cR491-R492

Reply:
I've added tests to trigger this branch. I tested that they hit the branch by applying:

```diff
diff --git a/numba/cuda/cudaimpl.py b/numba/cuda/cudaimpl.py
index 7e6cb1832..6eb425f10 100644
--- a/numba/cuda/cudaimpl.py
+++ b/numba/cuda/cudaimpl.py
@@ -490,6 +490,7 @@ def round_to_impl(context, builder, sig, args):
                 pow2 = 1.0
             y = (x * pow1) * pow2
             if math.isinf(y):
+                print(11111)
                 return x
 
         else:
```

and checking that the test cases printed out `11111`.